### PR TITLE
[f] 063 - get handlers from handler module developer provided

### PR DESCRIPTION
**Pull request by giang271291**
Description
--
- Add `utils.handler_utils` to discover user's handlers
- User can pass the Python module path to some package containing their handlers when run the bot:

How to test
--
* Create a test project and install twitterbot locally by `pip install -e /path/to/twitterbot/repo`
* Put all your BaseTweetHandler derived classes inside some `handlers` package
* Run twitterbot
```
$ start_twitterbot --handlers-package handlers
```

### CODE REVIEW

File: twitterbot/utils/handler_utils.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+from twitterbot.handlers.base import BaseTweetHandler
+
+
+def discover_handler_classes(handlers_package):
+    """
+    Looks for handler classes within handler path module.
+
+    Currently it's not looking deep into nested module
+    :param handlers_package: module path to handlers
+    :return: list of handler classes
```
Comment (hotay) [inline twitterbot/utils/handler_utils.py]: Just curious, what format is this docstring in? Can it be useful in other tools?

Comment (anhhuy1605) [inline twitterbot/utils/handler_utils.py]: I used the PyCharm generated one, I think it's reStructuredText and we can use Sphinx to generate documentation from that

File: twitterbot/twitterbot.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+    def start(self, handlers_package=None, *args, **kwargs):
         print('Twitterbot started')
-        # TODO: Find handlers here
-        handlers = []
 
-        # TODO: Select which wrapper to use, e.g. Utils.get_api('tweepy')
-        api = TweepyUtils.get_api('', '', '', '')
+        handler_classes = handler_utils.discover_handler_classes(handlers_package)
+        if not handler_classes:
+            print('No handler found. Did you forget to extend BaseTweethandler? Check --handlers-module')
```
Comment (tklarryonline) [inline twitterbot/twitterbot.py]: How about using Click's `.echo()` function? Does it help with the print in Py2 and 3?

**Comment by tklarryonline** [comment]
LGTM :)

File: twitterbot/twitterbot.py [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
-    def start(self):
+    def start(self, handlers_package=None, *args, **kwargs):
         print('Twitterbot started')
-        # TODO: Find handlers here
-        handlers = []
 
-        # TODO: Select which wrapper to use, e.g. Utils.get_api('tweepy')
-        api = TweepyUtils.get_api('', '', '', '')
+        handler_classes = handler_utils.discover_handler_classes(handlers_package)
+        if not handler_classes:
```
Comment (tklarryonline) [inline twitterbot/twitterbot.py]: I would prefer `discover_handler_classes` to raise an Exception in this case, and let Click handles the abort.

File: twitterbot/twitterbot.py (hunk 2 above)
Comment (giang271291) [inline twitterbot/twitterbot.py]: `print()` works with python2, so for now I don't think we need to replace the familiar `print()` function with `click.echo()` yet

File: twitterbot/twitterbot.py (hunk 3 above)
Comment (giang271291) [inline twitterbot/twitterbot.py]: @tklarryonline We would like to call the bot from other programs also. In that case, a generic termination method is better than Click's (which is specifically for programs invoked from Terminal)

File: twitterbot/twitterbot.py (hunk 2 above)
Comment (anhhuy1605) [inline twitterbot/twitterbot.py]: `print` is just a placeholder for log, we will replace them when we work on logging feature

Comment (tklarryonline) [inline twitterbot/twitterbot.py]: Reasonable for me.

File: twitterbot/twitterbot.py (hunk 3 above)
Comment (tklarryonline) [inline twitterbot/twitterbot.py]: I think a generic termination method is to raise an exception, though. Other developers can utilize the error we produced and act upon it then.

**Comment by tklarryonline** [comment]
I've got this error when trying this: `start_twitterbot --handlers-package examples.handlers.handler`

```
AttributeError: module 'examples.handlers.handler' has no attribute '__path__'
```

Wonder what happened?

**Comment by tklarryonline** [comment]
LGTM now :D

**Pull request merged**